ml-agents/mlagents/trainers/settings.py

```diff
@@ -292,6 +292,13 @@ def apply(self, key: str, env_channel: EnvironmentParametersChannel) -> None:
         """
         pass
 
+    @abc.abstractmethod
+    def to_string(self) -> str:
+        """
+        Helper method to output sampler stats to console.
+        """
+        pass
```

Comment:
I think Python has a default __str__ method that we could override.
Also, I think this should raise a NotImplementedError rather than just pass, what do you think?

Reply:
updated to this.